File: Framework/Java/framework/framework/smartkernel/framework/net/SmartMailInfo.java
/*
 * 项目：Smartkernel.Framework
 * 作者：曹艳白
 * 邮箱：caoyanbai@139.com
 */
package smartkernel.framework.net;

import java.util.*;

/**
 * 邮件内容实体类
 *
 */
public class SmartMailInfo {
	private String htmlBody;

	private String from;

	private Date date;

	private String subject;

	/**
	 * 获取邮件正文
	 * 
	 * @return，邮件正文
	 */
	public String getHtmlBody() {
		return htmlBody;
	}

	/**
	 * 设置邮件正文
	 * 
	 * @param htmlBody
	 *            ，邮件正文
	 */
	public void setHtmlBody(String htmlBody) {
		this.htmlBody = htmlBody;
	}

	/**
	 * 获取邮件来源地址
	 * 
	 * @return，邮件来源地址
	 */
	public String getFrom() {
		return from;
	}

	/**
	 * 设置邮件来源地址
	 * 
	 * @param from
	 *            ，邮件来源地址
	 */
	public void setFrom(String from) {
		this.from = from;
	}

	/**
	 * 获取邮件发送时间
	 * 
	 * @return，邮件发送时间
	 */
	public Date getDate() {
		return date;
	}

	/**
	 * 设置邮件发送时间
	 * 
	 * @param date
	 *            ，邮件发送时间
	 */
	public void setDate(Date date) {
		this.date = date;
	}

	/**
	 * 获取邮件标题
	 * 
	 * @return，邮件标题
	 */
	public String getSubject() {
		return subject;
	}

	/**
	 * 设置邮件标题
	 * 
	 * @param subject
	 *            ，邮件标题
	 */
	public void setSubject(String subject) {
		this.subject = subject;
	}
}

File: Framework/CSharp/Framework/Framework/Net/SmartMailInfo.cs
/*
 * 项目：Smartkernel.Framework
 * 作者：曹艳白
 * 邮箱：caoyanbai@139.com
 */
using System;
using LumiSoft.Net.Mail;
using LumiSoft.Net.MIME;

namespace Smartkernel.Framework.Net
{
    /// <summary>
    /// 邮件内容实体类
    /// </summary>
    public class SmartMailInfo
    {
        private readonly Mail_Message mime;

        /// <summary>
        /// 构造函数
        /// </summary>
        /// <param name="mime">Mime类型</param>
        public SmartMailInfo(Mail_Message mime)
        {
            this.mime = mime;
        }

        /// <summary>
        /// 获取邮件正文
        /// </summary>
        public string HtmlBody { get { return mime.BodyHtmlText; } }

        /// <summary>
        /// 获取邮件来源地址
        /// </summary>
        public string From { get { return mime.From.ToString(); } }

        /// <summary>
        /// 获取邮件的附件
        /// </summary>
        public MIME_Entity[] Attachments { get { return mime.Attachments; } }

        /// <summary>
        /// 获取邮件发送时间
        /// </summary>
        public DateTime Date { get { return mime.Date; } }

        /// <summary>
        /// 获取邮件标题
        /// </summary>
        public string Subject { get { return mime.Subject; } }
    }
}